src/tools/validations.ts

```diff
@@ -1,9 +1,50 @@
-import { GluegunToolbox } from "gluegun"
+import { GluegunToolbox, strings } from "gluegun"
 import { prefix } from "./pretty"
+const { pascalCase } = strings
+
+// #region Error Guards
+type IsError = (str: string) => boolean
+type ErrorMessage = (str?: string) => string
+type ErrorGuard = [IsError, ErrorMessage]
+
+const isIgnite: ErrorGuard = [
+  (str) => str.toLowerCase() === "ignite",
+  (str) => `Hey...that's my name! Please name your project something other than '${str}'.`,
+]
+const isOnlyNumbers: ErrorGuard = [
+  (str) => /^\d+$/.test(str),
+  () => `Please use at least one non-numeric character for your project name`,
+]
+const isKebabCase: ErrorGuard = [
+  (str) => str.includes("-"),
+  (str) =>
+    `Please use 'PascalCase', not 'kebab-case' for your project name. Ex: ${pascalCase(str)}`,
+]
+const isNotAlphaNumeric: ErrorGuard = [
+  (str) => !/^[a-z_][a-z0-9_]+$/i.test(str),
+  () =>
+    `The project name can only contain alphanumeric characters and underscore, but must not begin with a number.`,
+]
```

Comment:
All of these guard function are the same checks as the existing conditionals in `validateProjectName`

However, the advantage is that we can use these guards now both for the enquirer prompt and cli parameter